pkg/test/context.go

```diff
@@ -85,3 +100,40 @@ func (ctx *TestCtx) CleanupNoT() {
 func (ctx *TestCtx) AddFinalizerFn(fn finalizerFn) {
 	ctx.CleanUpFns = append(ctx.CleanUpFns, fn)
 }
+
+// TODO: figure out how to properly retrieve object kind info from object and enable
+// commented out logging
+
+// CreateWithFinalizer uses the dynamic client to create an object and then adds a
+// finalizer function to delete it when Cleanup is called. In addition to the standard
+// controller-runtime client options
+func (ctx *TestCtx) CreateWithFinalizer(gCtx goctx.Context, obj runtime.Object, cleanupOptions *CleanupOptions) error {
+	err := Global.DynamicClient.Create(gCtx, obj)
+	if err != nil {
+		return err
+	}
+	key, err := dynclient.ObjectKeyFromObject(obj)
+	//ctx.t.Logf("resource type %+v with namespace/name \"%+v\" created\n", obj.GetObjectKind(), key)
+	ctx.AddFinalizerFn(func() error {
+		err = Global.DynamicClient.Delete(gCtx, obj)
+		if err != nil {
+			return err
+		}
+		if cleanupOptions != nil && !cleanupOptions.SkipPolling {
+			return wait.PollImmediate(time.Second*1, time.Second*5, func() (bool, error) {
```

Comment:
are we supposed to use the `cleanupOptions.Timeout ` and `cleanupOptions.finalizerRetryInterval` for `wait.PollImmediate(...)`?

Reply:
Oops, yeah, forgot to change that when I added cleanupOptions

Reply:
Fixed